core/src/components/button/button.tsx

```diff
@@ -40,6 +41,8 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
 
   @Element() el!: HTMLElement;
 
+  @State() isCircle: boolean = false;
```

Comment:
Is the circle button required for this initial work? We have in progress work to implement this: https://github.com/ionic-team/ionic-framework/pull/29161.

I believe later we could add the circle button implementation for the ionic theme, after the feature is approved/merged.

Reply:
I did already that question as well, but it was at the design docs and also at the button requirements list, that's way I implemented it :/

Reply:
I would say to keep it for now since we've at the requirements and design doc as well. Once we've that clear we can remove it and use the new one if that's the expected ;)

Reply:
The `next` branch has been updated to include the circle button. So we can resolve this conversation.